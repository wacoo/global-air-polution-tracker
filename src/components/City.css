.cell-detail {
  height: 150px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 12px;
  color: #fff;
}

.cell-detail:nth-child(even) {
  background-color: #004d00;
}

.cell-detail:nth-child(odd) {
  background-color: #022b02;
}

.cell-detail h2 {
  display: inline;
}

.data {
  font-family: Arial, Helvetica, sans-serif;
  font-weight: 500;
  font-size: 14px;
}

.city-sub {
  width: 100%;
}

.city-main img {
  width: 150px;
}

.city-main {
  width: 100%;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #267c26;
}

.city-main i {
  margin: 12px;
  font-size: 30px;
  color: #fff;
}

.city-main div {
  display: flex;
  width: 100%;
  height: 150px;
  background-color: #267c26;
  margin: 0 12px;
  align-items: center;
  justify-content: space-around;
}

.city-main h2 {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 25px;
  color: #fff;
}
