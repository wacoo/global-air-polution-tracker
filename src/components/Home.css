.container {
    width: 100%;
    height: 100vh;
    background-color: brown;
}

.main-area {
    width: 100%;
    height: 30%;
    background-color: aqua;
}

.sub-area {
    width: 100%;
    height: 70vh;
    background-color: blueviolet;
}

.sub-row {
    width: 100%;
    height: 30%;
    background-color: antiquewhite;
    border: black solid 2px;
    display: flex;
}

.cell {
    width: 50%;
    height: 100%;
    border: purple solid 2px;
}