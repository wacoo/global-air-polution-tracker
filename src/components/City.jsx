import { useSelector } from 'react-redux';
import './City.css';
import { v4 as uuidv4 } from 'uuid';

const City = () => {
  const selectedCountry = useSelector(
    (state) => state.pollution.selectedCountry,
  );

  const uuid = uuidv4();
  const { capital } = selectedCountry;
  const {
    co, no, nh3, o3, pm10, pm2_5: pm2, so2,
  } = capital.pollution;

  return (
    <div className="container">
      <div className="city-main">
        <a href="/">
          <i className="fa fa-arrow-circle-left" />
        </a>
        <div>
          <h2>{selectedCountry.name.common}</h2>
          <img src={selectedCountry.flags.png} alt="" />
        </div>
      </div>
      <div className="city-sub">
        <div className="cell-detail" key={uuid}>
          <h2>{capital.name}</h2>
          <div className="data">
            <span>{`CO: ${co} NO: ${no}`}</span>
            <br />
            <span>{`NH3: ${nh3} O3: ${o3}`}</span>
            <br />
            <span>{`PM10: ${pm10} PM2_5: ${pm2}`}</span>
            <br />
            <span>{`SO2: ${so2}`}</span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default City;
