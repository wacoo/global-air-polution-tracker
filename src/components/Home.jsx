import './Home.css';
import Country from './Country';
import { useSelector, useDispatch } from 'react-redux';
import { useEffect, useState } from 'react';
import { clearState, fetchCountryData, getFiltered, setFetchType, setHistoryDateUnix } from '../redux/pollution/pollutionSlice';
import globalData from '../data/gps';

const Home = () => {
    const pollution = useSelector((state) => state.pollution);
    const dispatch = useDispatch();

    const filterCountry = (value) => {
        dispatch(getFiltered(value));
    }

    return (
        <div className="container">
          <div className="main-area">
            <input type="text" placeholder="Search..." onChange={(e) => filterCountry(e.target.value)} />           
          </div>
          <div className="sub-area">
            <Country />
          </div>
        </div>
      );
}

export default Home;