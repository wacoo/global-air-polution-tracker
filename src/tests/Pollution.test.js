import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { useDispatch, useSelector } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Home from '../components/Home';
import City from '../components/City';
import '@testing-library/jest-dom/extend-expect';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

const fakeData = {
  countries: [
    {
      name: 'United States',
      continent: 'North America',
      gps: [37.0902, -95.7129],
      map: '',
      cities: [
        {
          name: 'New York City',
          gps: [40.7128, -74.006],
        },
        {
          name: 'Los Angeles',
          gps: [34.0522, -118.2437],
        },
      ],
      pollution: {
        co: '200',
        no: '30',
        nh3: '0.5',
        o3: '600',
        pm10: '12',
        pm2_5: '0.6',
        so2: '23',
      },
    },
    {
      name: 'Canada',
      continent: 'North America',
      gps: [56.1304, -106.3468],
      map: '',
      cities: [
        {
          name: 'Toronto',
          gps: [43.65107, -79.347015],
        },
        {
          name: 'Vancouver',
          gps: [49.2827, -123.1207],
        },
      ],
      pollution: {
        co: '200',
        no: '30',
        nh3: '0.5',
        o3: '600',
        pm10: '12',
        pm2_5: '0.6',
        so2: '23',
      },
    },
    {
      name: 'Mexico',
      continent: 'North America',
      gps: [23.6345, -102.5528],
      map: '',
      cities: [
        {
          name: 'Mexico City',
          gps: [19.4326, -99.1332],
        },
        {
          name: 'Cancun',
          gps: [21.1619, -86.8515],
        },
        {
          name: 'Guadalajara',
          gps: [20.6597, -103.3496],
        },
        {
          name: 'Monterrey',
          gps: [25.6866, -100.3161],
        },
        {
          name: 'Tijuana',
          gps: [32.5149, -117.0382],
        },
      ],
      pollution: {
        co: '200',
        no: '30',
        nh3: '0.5',
        o3: '600',
        pm10: '12',
        pm2_5: '0.6',
        so2: '23',
      },
    },
  ],
};

describe('Test components render', () => {
  it('Tests Home', () => {
    useDispatch();
    useSelector.mockReturnValue(fakeData);
    const homeComponent = renderer.create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>,
    );
    expect(homeComponent).toMatchSnapshot();
  });

  it('Tests City', () => {
    useDispatch();
    useSelector.mockReturnValue(fakeData.countries[0]);
    const homeComponent = renderer.create(
      <BrowserRouter>
        <City />
      </BrowserRouter>,
    );
    expect(homeComponent).toMatchSnapshot();
  });
});

describe('Test existance of elements in components', () => {
  it('Tests existance of the button Mexico in Home ', () => {
    useDispatch();
    useSelector.mockReturnValue(fakeData);
    render(<Home />);
    const h2 = screen.getByText('Mexico');
    expect(h2).toBeInTheDocument();
  });
  it('Tests existance of New York City entry in datail page ', () => {
    useDispatch();
    useSelector.mockReturnValue(fakeData.countries[0]);
    render(<City />);
    const h2 = screen.getByText('New York City');
    expect(h2).toBeInTheDocument();
  });
});
