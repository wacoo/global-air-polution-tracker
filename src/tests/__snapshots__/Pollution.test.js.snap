// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test components render Tests City 1`] = `
<div
  className="container"
>
  <div
    className="city-main"
  >
    <a
      href="/"
    >
      <i
        className="fa fa-arrow-circle-left"
      />
    </a>
    <div>
      <h2>
        United States
      </h2>
      <img
        alt=""
        src=""
      />
    </div>
  </div>
  <div
    className="city-sub"
  >
    <div
      className="cell-detail"
    >
      <h2>
        New York City
      </h2>
      <div
        className="data"
      >
        <span>
          CO:  NO: 
        </span>
        <br />
        <span>
          NH3: 0 O3: 
        </span>
        <br />
        <span>
          PM10:  PM2_5: 
        </span>
        <br />
        <span>
          SO2: 
        </span>
      </div>
    </div>
    <div
      className="cell-detail"
    >
      <h2>
        Los Angeles
      </h2>
      <div
        className="data"
      >
        <span>
          CO:  NO: 
        </span>
        <br />
        <span>
          NH3: 0 O3: 
        </span>
        <br />
        <span>
          PM10:  PM2_5: 
        </span>
        <br />
        <span>
          SO2: 
        </span>
      </div>
    </div>
    ;
  </div>
</div>
`;

exports[`Test components render Tests Home 1`] = `
<div
  className="container"
>
  <div
    className="main-area"
  >
    <input
      onChange={[Function]}
      placeholder="Search..."
      type="text"
    />
  </div>
  <div
    className="sub-area"
  >
    <button
      className="cell"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url()",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
        }
      }
      type="button"
    >
      <div
        className="inside-icon-wrapper"
      >
        <i
          className="fa fa-arrow-circle-right"
        />
      </div>
      <div
        className=""
      >
        <h2>
          United States
        </h2>
        <div
          className="pullution"
        >
          CO: 200 NO: 30
          <br />
          NO2: undefined NH3: 0.5
          <br />
          O3: 600 PM10: 12
          <br />
          PM2_5: 0.6 SO2: 23
        </div>
      </div>
    </button>
    <button
      className="cell"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url()",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
        }
      }
      type="button"
    >
      <div
        className="inside-icon-wrapper"
      >
        <i
          className="fa fa-arrow-circle-right"
        />
      </div>
      <div
        className=""
      >
        <h2>
          Canada
        </h2>
        <div
          className="pullution"
        >
          CO: 200 NO: 30
          <br />
          NO2: undefined NH3: 0.5
          <br />
          O3: 600 PM10: 12
          <br />
          PM2_5: 0.6 SO2: 23
        </div>
      </div>
    </button>
    <button
      className="cell"
      onClick={[Function]}
      style={
        Object {
          "backgroundImage": "url()",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "contain",
        }
      }
      type="button"
    >
      <div
        className="inside-icon-wrapper"
      >
        <i
          className="fa fa-arrow-circle-right"
        />
      </div>
      <div
        className=""
      >
        <h2>
          Mexico
        </h2>
        <div
          className="pullution"
        >
          CO: 200 NO: 30
          <br />
          NO2: undefined NH3: 0.5
          <br />
          O3: 600 PM10: 12
          <br />
          PM2_5: 0.6 SO2: 23
        </div>
      </div>
    </button>
  </div>
</div>
`;
